drivers: counter: max32: Fix prescaler configuration

ttmut opened the pull request:
Fixes a bug in the prescaler configuration formula. Prescaler enumerations are not consecutive numbers, so the existing formula does not work as expected. <s>Add prescaler enums into a lookup table and set counter prescaler from this array using prescaler_index parameter.</s>

<s>This is the same fix introduced in https://github.com/zephyrproject-rtos/zephyr/pull/95409, but for the counter driver.</s>

hfakkiz committed:
drivers: counter: max32: Fix prescaler configuration

The existing formula assumed prescaler enumerations were consecutive,
which led to incorrect values being passed to HAL functions. Update
the code to calculate correct prescaler enumerations.

Signed-off-by: Furkan Akkiz <hasanfurkan.akkiz@analog.com>

MaureenHelm reviewed (commented):
This part of the commit message needs to be updated or removed:
```
Add prescaler enums into a lookup table and set
counter prescaler from this array using prescaler_index parameter.
```
Otherwise LGTM

ttmut committed:
drivers: pwm: max32: Simplify prescaler calculation

Replace the look-up table used to obtain the prescaler enumeration with
a simple multiplication.

Signed-off-by: Tahsin Mutlugun <Tahsin.Mutlugun@analog.com>
